packages/@react-aria/utils/src/scrollIntoView.ts

```diff
@@ -10,32 +10,68 @@
  * governing permissions and limitations under the License.
  */
 
+import {getScrollParent} from './';
+
+let isScrollPrevented = false;
+
+export function setScrollPrevented(value: boolean) {
```

Comment:
A bit hesitant about exposing this global state

Reply:
Have any ideas? We can't invert it so that the popover state of isPositioned is used, the dependency already is popover depends on utils, not utils depending on popover.

Can't use context because it's the aria hooks that are calling these and it seems worse to pass this into the hooks.

If we watched for attributes in the dom, we'd be too delayed since those are async fired.

We could try some different names that are less about the state we're setting and more about the action that's happening? popoverIsPositioning? not sure what else to do about scrollingPrevented though...

Reply:
Thanks for the brainstormed alternatives, I actually hadn't considered the DOM attributes one.

> We could try some different names that are less about the state we're setting and more about the action that's happening? popoverIsPositioning? not sure what else to do about scrollingPrevented though...

IMO, I'd prefer keeping it about the state we are setting since I'm uncertain what else might call `usePreventScroll` in the future.

Reply:
could you check `isScrollable(document.scrollingElement)`?

Reply:
Unfortunately, `isScrollable(document.scrollingElement)` returns false even if the page is scrollable. The `scrollingElement` (aka <html>) could have `overflow: visible` but a child div/element might have `overflow: auto`/`overflow: scroll` (aka body/root div/etc) and thus be scrollable. 

I guess I could query if `document.scrollingElement/documentElement` has `overflow: hidden` applied on it instead then, hmm

Reply:
Pushed the update, checking for `overflow: hidden` works for our standard and mobile safari prevent scroll strategies. Note that there is still a reliance on delaying the `scrollIntoViewport` call until the popover is fully positioned (aka `usePreventScroll` is delayed in `usePopover` until positioning is finished)